crates/core/app/src/params/change.rs

```diff
@@ -1,68 +1,92 @@
 use std::fmt::Display;
 
 use anyhow::Result;
+use penumbra_chain::params::{ChainParameters, Ratio};
+use penumbra_dao::params::DaoParameters;
+use penumbra_governance::params::GovernanceParameters;
+use penumbra_ibc::params::IBCParameters;
+use penumbra_stake::params::StakeParameters;
 
-use super::{ChainParameters, Ratio};
+use super::AppParameters;
 
 // The checks below validate that a parameter change is valid, since some parameter settings or
 // combinations are nonsensical and should be rejected outright, regardless of governance.
 
 #[deny(unused)] // We want to be really careful here to not examine fields!
-impl ChainParameters {
-    pub fn check_valid_update(&self, new: &ChainParameters) -> Result<()> {
+impl AppParameters {
+    pub fn check_valid_update(&self, new: &AppParameters) -> Result<()> {
         new.check_valid()?;
+        // TODO: move the checks below into their respective components
```

Comment:
👍 commenting to flag something we should track as a follow-up. Probably we could do it while updating the parameter change logic?